src/ui/hook/useTimeout.ts

```diff
@@ -0,0 +1,29 @@
+import { useCallback, useEffect, useRef } from 'react'
+
+type ClearTimeout = () => void
+type ResetTimeout = () => void
+type TimeoutMethods = [ClearTimeout, ResetTimeout]
```

Comment:
```suggestion
type UseTimeout = [ClearTimeout, ResetTimeout]
```

More conventional and explicit ;)